src/codecs/md/__fixtures__/tableChunkFigure.md

```diff
@@ -0,0 +1,13 @@
+chunkfigure: Figure 1A
+:::
+| A | B                                        | C |
+| - | ---------------------------------------- | - |
+| 1 | Some content split accross multiple rows | 3 |
+| 4 | Some content split accross multiple rows | 6 |
+---
+## Classification and variability of neuronal cell types
+
+(**A**) Neuronal cell types are identifiable by features clustering around a distinct point (blue, green and yellow) or a line (orange) in feature space. The clustering implies that neuron types are defined by either a single set point (blue, green and yellow) or by multiple set points spread along a line (orange).
+:::
+
+{#fig1a}
```

Comment:
I think you may have misinterpreted what I meant by

> add decodeCodeChunkTable and encodeCodeChunkTable functions; I suspect we will need to have a table block extension which combines a Markdown table and a structured caption (that is analogous to how the current figure block extension combines a Markdown image and a structured caption) as well as a chunktable extension.

So to try to clarify what I meant. First, I think we need a new `table` block extension (table + caption) that is analogous to the `figure` block extension (image + caption). This would use the common convention of having the caption above the table. An example:

```
table: Table 1
:::
## Classification and variability of neuronal cell types

Neuronal cell types are identifiable by features clustering around a distinct point (blue, green and yellow) or a line (orange) in feature space. The clustering implies that neuron types are defined by either a single set point (blue, green and yellow) or by multiple set points spread along a line (orange).
---
| A | B                                        | C |
| - | ---------------------------------------- | - |
| 1 | Some content split across multiple rows | 3 |
| 4 | Some content split across multiple rows | 6 |
:::
{#table1}
````

Second, we need a `chunktable` extension which allows for tables that are generated by a code chunk e.g.

````
chunktable: Table 1
:::
## Classification and variability of neuronal cell types

Neuronal cell types are identifiable by features clustering around a distinct point (blue, green and yellow) or a line (orange) in feature space. The clustering implies that neuron types are defined by either a single set point (blue, green and yellow) or by multiple set points spread along a line (orange).
---
```r
head(data)
```
| A | B                                        | C |
| - | ---------------------------------------- | - |
| 1 | Some content split across multiple rows | 3 |
| 4 | Some content split across multiple rows | 6 |
:::
{#table1}
````

Note that this example encodes an executed `CodeChunk` and therefore the `output` table is also encoded to Markdown. If the `CodeChunk` was not wrapped with a caption, it would be encoded in Markdown as a plain `codechunk`:

````
codechunk:
:::
```r
head(data)
```
| A | B                                        | C |
| - | ---------------------------------------- | - |
| 1 | Some content split across multiple rows | 3 |
| 4 | Some content split across multiple rows | 6 |
:::
````

Reply:
Thanks for clarifying @nokome.
As I was working on this I had two inter-related questions that I would appreciate your input on.

1. To keep the authoring experience simpler, what do you think about having a single `table` block extension and handling splitting of the code and output in the decoding step?

2. The `chunktable` example to me seems like it is should instead be a CodeChunk (both semantically and structurally). This would require adding `caption` and `label` fields to the schema though, with the generated table captured in the `outputs`. Otherwise I don't think we can capture the relationship of the generated table and the code in the schema, but I might be missing something.

Reply:
> adding caption and label fields to the schema though, with the generated table captured in the outputs

I really like this idea. It would mean that we would only need three block extensions: `figure`, `table` and `chunk`. The `figure` and `table` extensions would just serve to wrap together static `ImageObject` and `Table` nodes with their captions and labels. The `chunk` could be used for wither a table or a image that was produced by code.

A chunk used for a table:

````markdown
chunk: Table 1
:::
```r
head(data)
```
| A | B                                        | C |
| - | ---------------------------------------- | - |
| 1 | Some content split across multiple rows | 3 |
| 4 | Some content split across multiple rows | 6 |
---
## Classification and variability of neuronal cell types

Neuronal cell types are identifiable by features clustering around a distinct point (blue, green and yellow) or a line (orange) in feature space. The clustering implies that neuron types are defined by either a single set point (blue, green and yellow) or by multiple set points spread along a line (orange).
:::
{#table1}
````


A chunk used for a figure:

````markdown
chunk: Figure 1
:::
```r
plot(data)
```
![](article.rmd.media/fig1.png)
---
## Classification and variability of neuronal cell types

Neuronal cell types are identifiable by features clustering around a distinct point (blue, green and yellow) or a line (orange) in feature space. The clustering implies that neuron types are defined by either a single set point (blue, green and yellow) or by multiple set points spread along a line (orange).
:::
{#fig1}
````

> what do you think about having a single table block extension and handling splitting of the code and output in the decoding step?

Rather than doing that, I'd go for the idea of keeping `table` for static tables and `chunk` for dynamic tables or figures as above.